### PR TITLE
handle negotiate challenge

### PR DESCRIPTION
**Pull request by njarecha**

### CODE REVIEW

**Review by dfeinzimer — APPROVED** [review]
Thanks @njarecha!

File: Sources/ArcGISToolkit/Components/Authentication/NetworkChallengeContinuation.swift [hunk 1]
```diff
@@ -72,6 +72,8 @@ extension NetworkChallengeContinuation.Kind {
             self = .login
         case .clientCertificate:
             self = .certificate
+        case .negotiate:
+            fatalError("Negotiate challenge continuation is not implemented.")
```
Comment (rolson) [inline Sources/ArcGISToolkit/Components/Authentication/NetworkChallengeContinuation.swift:76]: To me this sounds like it needs to happen at some point. But if that's not the case I think we might want a more clear message. Is this accurate?
```suggestion
            fatalError("We should not end up here as negotiate challenges are rejected when they come in.")
```

Comment (rolson) [reply Sources/ArcGISToolkit/Components/Authentication/NetworkChallengeContinuation.swift:76]: The other option is to make the init failable.

Comment (njarecha) [reply Sources/ArcGISToolkit/Components/Authentication/NetworkChallengeContinuation.swift:76]: Updated message looks good.

**Review by rolson — APPROVED** [review]

**Review by dfeinzimer — APPROVED** [review]

**Comment by njarecha** [comment]
Thanks @dfeinzimer and @rolson!

**Pull request merged**